### ISSUE TITLE
Problem running the bootstrap script 

### ISSUE THREAD

**Issue by carnageX** [TykTechnologies/tyk#148]
I've setup a VM (8GB virtual drive, set to be dynamic) of Ubuntu Server LTS 14.04 to try out Tyk.  I get up to the point of running the bootstrap.sh script, and it returns with this:

Creating Organisation
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/json/__init__.py", line 290, in load
    **kw)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
ORGID:
Adding new user
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/json/__init__.py", line 290, in load
    **kw)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
USER AUTH:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/json/__init__.py", line 290, in load
    **kw)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
NEW ID:
Setting password

DONE

Thus, I'm unable to connect to the dashboard (presumably because it hasn't finished setup at some point... when I try and navigate to the URL it spits out at the end, I just get ERR_CONNECTION_REFUSED by my browser --- maybe a different or related issue?).  

Thanks for any help, I appreciate it.  

Edit: these are the instructions I was following
https://tyk.io/v1.9/setup/install-tyk-ubuntu/

**Comment by lonelycode** [comment]
When the bootstrap script does that, dashboard is not running.

Check if dashboard is running using 

sudo service tyk-dashboard status

If it is running - did you go through the configuration (setup) step and change the port number for dashboard? What port is it running on?

If you can't browse to the URL then dashboard is definitely not running.

**Comment by carnageX** [comment]
Yep, the dashboard is indeed running.  

It should be running on 3000. 

Here's part of my tyk_analytics.conf file
{
    "listen_port": 3000,
    "tyk_api_config": {
        "Host": "http://localhost",
        "Port": "8080",
        "Secret": "352d20ee67be67f6340b4c0605b044b7"
    },
    "mongo_url": "mongodb://localhost/tyk_analytics",
    "page_size": 10,
    "admin_secret": "12345",
    "redis_port": 6379,
    "redis_host": "localhost",
    "redis_password": "",
    "force_api_defaults": false,
    "notify_on_change": true,
    "license_owner": "Your Name",
    "hash_keys": true,
    "email_backend": {},
    "hide_listen_path": false,

**Comment by lonelycode** [comment]
if you can't browse to the server and get a connection refused error, then the dashboard might be running but that port is blocked. Are you running it in a VM?

If so, you need to open all the ports

if you cannot browse to the dashboard then the bootstrap script can't access it either (it uses the API to get things set up, there's no magic DB interactions)

Did you run the setup properly before starting the process?

If the process is running, then there will be a log file in car/log/upstart - maybe check that it to see if anything is up.

If you can't get to the login screen, you can't bootstrap it, so solving that problem is your main issue, and if it's running then it's the OS, not the process.

**Comment by carnageX** [comment]
Yes, I'm running it in a VM.  However I can't even get to it via localhost on the VM itself. 

I assume I ran the setup properly...the only instructions I saw was the set that I previously posted (https://tyk.io/v1.9/setup/install-tyk-ubuntu/).  

This is what my tyk-dashboard.log file looks like:

^[[31mERRO^[[0m[0253] Mongo connection failed (retrying):no reachable servers
^[[34mINFO^[[0m[0000] Using /opt/tyk-dashboard/tyk_analytics.conf for configuration
^[[31mERRO^[[0m[0011] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0023] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0034] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0046] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0057] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0069] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0080] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0092] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0103] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0115] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0126] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0138] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0150] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0161] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0173] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0184] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0196] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0207] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0219] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0230] Mongo connection failed (retrying):no reachable servers
^[[31mERRO^[[0m[0242] Mongo connection failed (retrying):no reachable servers

**Comment by lonelycode** [comment]
Well that seems pretty clear - the dashboard can't start because the mongo DB can't be found :-) 

Is mongodb up and running?

**Comment by carnageX** [comment]
When I ran
sudo service mongod status 
it said it was: stop/waiting

Tried starting it with
sudo service mongod start
It said: start/running

But when I check the status again, it just goes back to stop/waiting

**Comment by lonelycode** [comment]
Would recommend taking a look at the mongodb docs on installing mongo on Ubuntu, there may be something with your distro/image.

You could always just create the required directories :-)

Sorry I can't help more, but I've never seen mongodb not install correctly - maybe you installed the wrong version or the installation just didn't work

You could try a hosted mongo if necessary. 

How are you virtualising?

**Comment by carnageX** [comment]
I don't imagine I installed the wrong version, unless the documentation is incorrect?  Possible that the install didn't work...but I tried on 2 different VM's of Ubuntu (both v14 and v15).  

I'll try creating the directories - found some things on SO that I may try.  

I'm hosting the VM in VirtualBox on a PC in my home to test out Tyk - nothing really special about it.

**Comment by lonelycode** [comment]
i don't think it's the OS, I think it's what it's running on, Mongo is trying to access files that don't exist but should because they would have been created on install.

Which means the install can't do something - if you are running in VirtualBox, with an 8GB disk, is that full? Is it read only?

If you are virtualising and trying to run Tyk - you could use Vagrant - it works with VirtualBox without all the manual setup (it also handles all your networking and disk allocation). We also have a setup guide for vagrant :-)

When you install mongo, are there any error messages? When we use Mongo in docker for example, it throws a fit if it doesn't have enough disk space - and this only on OSX because the way docker runs in OSX is with a VM and the VM doesn't have enough disk space :-)

So I really think it's the virtualisation & Mongo - maybe give your install a bigger drive?

**Comment by carnageX** [comment]
Just checked, and the disk is only 50%  used up; 2.9GB used, with currently allocated of 5.8GB (up to 8GB, will grow dynamically from VirtualBox).  So I wouldn't think it's that...unless Mongo requires a larger DB? 

I could look into Vagrant - but was wanting to demo this for a possible solution to use at my workplace - and not sure how well Vagrant would go over with the server people - I'm sure I'll have enough trouble trying to convince them to spin up a Linux box if we choose to go with Tyk ;). 

I don't remember seeing any error messages while installing Mongo - everything had gone smoothly with no errors (at least reported through SSH / terminal) while going through the instructions.  

I ran across this post on SO: http://stackoverflow.com/questions/7948789/mongodb-mongod-complains-that-there-is-no-data-db-folder

Tried creating the folder manually and setting the permissions, and now here's what I get:
carnagex@Swordfish:/$ sudo grep mongo /etc/passwd
mongodb:x:109:65534::/home/mongodb:/bin/false
carnagex@Swordfish:/$ sudo chmod 0755 /data/db
carnagex@Swordfish:/$ sudo chown -R 109:65534 /data/db
carnagex@Swordfish:/$ mongod
2015-12-10T04:24:51.822-0600 E NETWORK  [initandlisten] Failed to unlink socket file /tmp/mongodb-27017.sock errno:1 Operation not permitted
2015-12-10T04:24:51.822-0600 I -        [initandlisten] Fatal Assertion 28578
2015-12-10T04:24:51.822-0600 I -        [initandlisten]

***aborting after fassert() failure
